packages/Checkbox/src/Checkbox.styles.js

```diff
@@ -0,0 +1,130 @@
+import { css } from "styled-components";
+import stylers from "@paprika/stylers";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import tokens from "@paprika/tokens";
+
+const integize = token => Number.parseInt(token, 10);
+
+const sizeMap = {
+  [ShirtSizes.SMALL]: 2,
+  [ShirtSizes.MEDIUM]: 2.5,
+  [ShirtSizes.LARGE]: 3,
+};
+
+const getSpacing = size => stylers.spacer(sizeMap[size]);
+const getSpacingInt = size => integize(getSpacing(size));
+
+const size = ({ size }) => getSpacing(size);
+const fontSizeInt = size => (getSpacingInt(size) / integize(tokens.space) - 3) * 2;
+const fontSizeValue = ({ size }) => `${stylers.fontSizeValue(fontSizeInt(size))}px`;
+
+const topOffset = size => (getSpacingInt(size) - integize(stylers.fontSizeValue()) * stylers.lineHeightValue(-1)) / 2;
+const isCheckboxBigger = size => topOffset(size) > 0;
+const labelPadding = ({ hasChildren, size }) => {
+  const top = isCheckboxBigger(size) ? `${topOffset(size)}px` : 0;
+  const left = hasChildren ? `${getSpacingInt(size) + integize(tokens.space)}px` : getSpacing(size);
+  return `${top} 0 ${top} ${left}`;
+};
+
+const checkerTop = size => (isCheckboxBigger ? "-1px" : `${Math.abs(topOffset(size) / 2)}px`);
+const checkBoxIconLeft = ({ size }) => `${getSpacingInt(size) / 2}px`;
+
+const checkboxStyles = css`
+  ${stylers.boxSizingStyles};
+  font-size: ${fontSizeValue};
+  line-height: ${({ hasChildren }) => (hasChildren ? stylers.lineHeightValue(-1) : "0")};
+  position: relative;
+
+  input[type="checkbox"] {
+    ${stylers.visuallyHidden};
+
+    &:focus + label::before {
+      box-shadow: ${tokens.highlight.active.noBorder.boxShadow};
+      border-color: ${tokens.highlight.active.noBorder.borderColor};
+    }
+
+    & + label {
+      cursor: pointer;
+      display: inline-block;
+      margin: 0;
+      min-height: ${size};
+      padding: ${labelPadding};
+      position: relative;
+    }
+
+    & + label::before,
+    & + label > .checkbox-icon {
+      position: absolute;
+      top: ${checkerTop};
+    }
+
+    & + label::before {
+      background: ${tokens.color.white};
+      border: 2px solid ${tokens.border.color};
+      border-radius: ${tokens.border.radius};
+      content: "";
+      height: ${size};
+      left: 0;
+      width: ${size};
+      z-index: 1;
+    }
+
+    & + label:hover {
+      &:before {
```

Comment:
could just do this on one line instead of two
`& + label:hover:before`
Also it should be `::before`